linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/NameGen.scala

```diff
@@ -107,7 +110,13 @@ private[emitter] final class NameGen {
   }
 
   def genName(name: LabelName): String = genNameGeneric(name, genLabelNameCache)
-  def genName(name: FieldName): String = genNameGeneric(name, genFieldNameCache)
+  def genName(name: SimpleFieldName): String = genNameGeneric(name, genSimpleFieldNameCache)
```

Comment:
Note to self: When do we use this? If it is just in the scope, maybe we should move the scope to use `genName` as well to avoid the additional memory cost of `genSimpleFieldNameCache`?